dspace-api/src/test/java/org/dspace/curate/CuratorTest.java

```diff
@@ -0,0 +1,413 @@
+        Curator result = instance.setReporter(reporter);
+        assertEquals(expResult, result);
+        // TODO review the generated test code and remove the default call to fail.
+        fail("The test case is a prototype.");
+    }
+*/
+
+    /**
+     * Test of setTransactionScope method, of class Curator.
+     */
+/*
+    @Test
+    public void testSetTransactionScope() {
+        System.out.println("setTransactionScope");
+        Curator.TxScope scope = null;
+        Curator instance = new Curator();
+        Curator expResult = null;
+        Curator result = instance.setTransactionScope(scope);
+        assertEquals(expResult, result);
+        // TODO review the generated test code and remove the default call to fail.
+        fail("The test case is a prototype.");
+    }
+*/
+
+    /**
+     * Test of curate method, of class Curator.
+     */
+/*
+    @Test
+    public void testCurate_Context_String() throws Exception {
+        System.out.println("curate");
+        String id = "";
+        Curator instance = new Curator();
+        instance.curate(context, id);
+        // TODO review the generated test code and remove the default call to fail.
+        fail("The test case is a prototype.");
+    }
+*/
+
+    /**
+     * Test of curate method, of class Curator.
+     * Currently this just tests task properties and run parameters.
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testCurate_DSpaceObject()
+            throws Exception {
+        System.out.println("curate");
+
+        final String TASK_NAME = "dummyTask";
+
+        // Configure the task to be run.
+        ConfigurationService cfg = kernelImpl.getConfigurationService();
+        cfg.setProperty("plugin.named.org.dspace.curate.CurationTask",
+                DummyTask.class.getName() + " = " + TASK_NAME);
+        cfg.setProperty(TASK_NAME + '.' + TASK_PROPERTY_NAME, TASK_PROPERTY_VALUE);
+
+        // Get and configure a Curator.
+        Curator instance = new Curator();
+        instance.setReporter("-"); // Send any report to standard out. FIXME when DS-3989 is merged
+        instance.addTask(TASK_NAME);
+
+        // Configure the run.
+        Map<String, String> parameters = new HashMap<>();
+        parameters.put(RUN_PARAMETER_NAME, RUN_PARAMETER_VALUE);
+        instance.addParameters(parameters);
+
+        // Curate the site.
+        DSpaceObject dso = SITE_SERVICE.findSite(context);
+        instance.curate(context, dso);
+
+        // Check the result.
+        System.out.format("Task %s result was '%s'%n",
+                TASK_NAME, instance.getResult(TASK_NAME));
+        System.out.format("Task %s status was %d%n",
+                TASK_NAME, instance.getStatus(TASK_NAME));
+        assertEquals("Unexpected task status",
+                Curator.CURATE_SUCCESS, instance.getStatus(TASK_NAME));
+        assertEquals("Wrong run parameter", RUN_PARAMETER_VALUE, runParameter);
```

Comment:
How does runParameter get set?  I expected the curation task to be queried for this value.